metaflow/metaflow_git.py

```diff
@@ -0,0 +1,165 @@
+    def find_git_on_windows():
+        """find the path to the git executable on Windows"""
+        # first see if git is in the path
+        try:
+            subprocess.check_output(["where", "/Q", "git"])
+            # if this command succeeded, git is in the path
+            return "git"
+        # catch the exception thrown if git was not found
+        except subprocess.CalledProcessError:
+            pass
+        # There are several locations where git.exe may be hiding
+        possible_locations = []
+        # look in program files for msysgit
+        if "PROGRAMFILES(X86)" in environ:
+            possible_locations.append(
+                "%s/Git/cmd/git.exe" % environ["PROGRAMFILES(X86)"]
+            )
+        if "PROGRAMFILES" in environ:
+            possible_locations.append("%s/Git/cmd/git.exe" % environ["PROGRAMFILES"])
+        # look for the GitHub version of git
+        if "LOCALAPPDATA" in environ:
+            github_dir = "%s/GitHub" % environ["LOCALAPPDATA"]
+            if path.isdir(github_dir):
+                for subdir in os.listdir(github_dir):
+                    if not subdir.startswith("PortableGit"):
+                        continue
+                    possible_locations.append(
+                        "%s/%s/bin/git.exe" % (github_dir, subdir)
+                    )
+        for possible_location in possible_locations:
+            if path.isfile(possible_location):
+                return possible_location
+        # git was not found
+        return "git"
+
+    GIT_COMMAND = find_git_on_windows()
+
+
+def _get_repo_url():
+    """Get the repository URL from git config"""
+    try:
+        result = subprocess.run(
+            [GIT_COMMAND, "config", "--get", "remote.origin.url"],
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        if result.returncode == 0:
+            url = result.stdout.strip()
+            # Convert SSH URLs to HTTPS for clickable links
+            if url.startswith("git@"):
+                parts = url.split(":")
+                if len(parts) == 2:
+                    domain = parts[0].replace("git@", "")
+                    repo_path = parts[1]
+                    url = f"https://{domain}/{repo_path}"
+            return url
+        return None
+    except (OSError, subprocess.SubprocessError):
+        return None
+
+
+def _get_branch_name():
+    """Get the current git branch name"""
+    try:
+        result = subprocess.run(
+            [GIT_COMMAND, "rev-parse", "--abbrev-ref", "HEAD"],
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        return result.stdout.strip() if result.returncode == 0 else None
+    except (OSError, subprocess.SubprocessError):
+        return None
+
+
+def _get_commit_sha():
+    """Get the current git commit SHA"""
+    try:
+        result = subprocess.run(
```

Comment:
I wonder if there is a way to combine all the subprocess calls into one single call since we call them all at once.